src/Compilers/CSharp/Portable/Compilation/CSharpDeterministicKeyBuilder.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class CSharpDeterministicKeyBuilder : DeterministicKeyBuilder
+    {
+        internal static readonly CSharpDeterministicKeyBuilder Instance = new();
+
+        private CSharpDeterministicKeyBuilder()
+        {
+
+        }
+
+        protected override void WriteCompilationOptionsCore(JsonWriter writer, CompilationOptions options)
+        {
+            if (options is not CSharpCompilationOptions csharpOptions)
+            {
+                throw new InvalidOperationException();
+            }
+
+            base.WriteCompilationOptionsCore(writer, options);
+
+            writer.Write("unsafe", csharpOptions.AllowUnsafe);
+            writer.Write("topLevelBinderFlags", csharpOptions.TopLevelBinderFlags);
+            writer.WriteKey("globalUsings");
```

Comment:
nit: just `usings`.  That's what the option is called.  i think it woudl be good to keep in sync.